specification/dartLangSpec.tex

```diff
@@ -15993,11 +15984,15 @@ \subsection{Await Expressions}
 % NOTICE: Removed the requirement that an error thrown by $e$ is caught in a
 % future. There is no reason $var x = e; await x;$ and $await e$ should behave
 % differently, and no implementation actually implemented it.
-If the run-time type of $o$ is a subtype of \code{Future<$T$>},
-then let $f$ be $o$;
-otherwise let $f$ be the result of creating
-a new object using the constructor \code{Future<$T$>.value()}
-with $o$ as its argument.
+Let $R$ be the run-time type of $o$.
+If $R$ implements \code{Future<$S$>} for some $S$
+(\ref{interfaceSuperinterfaces}) then
+it is a dynamic type error if $S$ is not a subtype of $T$.
+If $S$ is a subtype of $T$, let $f$ be $o$.
```

Comment:
I usually handle descriptions of errors (run-time or compile-time) slightly differently:
```suggestion
Otherwise
(\commentary{if no run-time error occurred}),
let $f$ be $o$.
```

We could do that for consistency. However, I also think this style is a bit safer to read (a bit harder to misunderstand), because it puts the emphasis on the error: "One case is an error, all other cases proceed as follows".

With 'If $S$ is a subtype of $T$', there's more room to think that there could be even more cases.